src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigWatcher.java

```diff
@@ -129,246 +133,226 @@ private synchronized void onInitialBuildConfigs(BuildConfigList buildConfigs) {
-        if (existingBuildRunPolicy != null
-                && !existingBuildRunPolicy.equals(buildConfigProjectProperty
-                        .getBuildRunPolicy())) {
-            maybeScheduleNext(job);
-        }
-    }
+	private void updateJob(WorkflowJob job, InputStream jobStream, String jobName, BuildConfig buildConfig,
+			String existingBuildRunPolicy, BuildConfigProjectProperty buildConfigProjectProperty) throws IOException {
+		Source source = new StreamSource(jobStream);
+		job.updateByXml(source);
+		job.save();
+		logger.info("Updated job " + jobName + " from BuildConfig " + NamespaceName.create(buildConfig)
+				+ " with revision: " + buildConfig.getMetadata().getResourceVersion());
+		if (existingBuildRunPolicy != null
+				&& !existingBuildRunPolicy.equals(buildConfigProjectProperty.getBuildRunPolicy())) {
+			maybeScheduleNext(job);
+		}
+	}
 
-    private void upsertJob(final BuildConfig buildConfig) throws Exception {
-        if (isPipelineStrategyBuildConfig(buildConfig)) {
-            // sync on intern of name should guarantee sync on same actual obj
-            synchronized (buildConfig.getMetadata().getUid().intern()) {
-                ACL.impersonate(ACL.SYSTEM,
-                        new NotReallyRoleSensitiveCallable<Void, Exception>() {
-                            @Override
-                            public Void call() throws Exception {
-                                String jobName = jenkinsJobName(buildConfig);
-                                WorkflowJob job = getJobFromBuildConfig(buildConfig);
-                                boolean newJob = job == null;
-                                if (newJob) {
-                                    job = new WorkflowJob(Jenkins
-                                            .getActiveInstance(), jobName);
-                                }
-                                BulkChange bk = new BulkChange(job);
+	private void upsertJob(final BuildConfig buildConfig) throws Exception {
+		if (isPipelineStrategyBuildConfig(buildConfig)) {
+			// sync on intern of name should guarantee sync on same actual obj
+			synchronized (buildConfig.getMetadata().getUid().intern()) {
+				ACL.impersonate(ACL.SYSTEM, new NotReallyRoleSensitiveCallable<Void, Exception>() {
+					@Override
+					public Void call() throws Exception {
+						String jobName = jenkinsJobName(buildConfig);
+						String jobFullName = jenkinsJobFullName(buildConfig);
+						WorkflowJob job = getJobFromBuildConfig(buildConfig);
+						Jenkins activeInstance = Jenkins.getActiveInstance();
+						ItemGroup parent = activeInstance;
+						if (job == null) {
+							job = (WorkflowJob) activeInstance.getItemByFullName(jobFullName);
+						}
+						boolean newJob = job == null;
+						if (newJob) {
+							String disableOn = getAnnotation(buildConfig, DISABLE_SYNC_CREATE);
+							if (disableOn != null && disableOn.equalsIgnoreCase("jenkins")) {
```

Comment:
@rupalibehera - I believe @bparees 's comment got lost in the shuffle of other changes; there is not reason the propr value has to be "jenkins"; let's simply go with the value len > 0; i.e. if set to anything disable, 
if it is not set, then it is enabled

Reply:
Yeah makes sense to me too, let me confirm with @jstrachan once if he has some other thoughts.